### PR TITLE
Fix for JENKINS-63983

### PR DESCRIPTION
**Pull request by rafeeqpv**
Tested with a Scriptler script containing the @Grapes annotation using Jenkins 2.289.2.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Allowed the approved Scriptler scripts to run outside sandbox mode.
- [x] https://issues.jenkins.io/browse/JENKINS-63983
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/biouno/unochoice/model/ScriptlerScript.java
Comment (kinow) [inline src/main/java/org/biouno/unochoice/model/ScriptlerScript.java:135]: A small fix with great benefit to users! Thanks @rafeeqpv , I will just need to i) check the issue to remind myself of the context, and ii) see if I can write a quick test so we don't have a regression in the future. Maybe iii) check if there's no security issue enabling it in Jenkins plug-ins. Will try to include this issue in the next release.

Comment (kinow) [reply src/main/java/org/biouno/unochoice/model/ScriptlerScript.java:135]: On iii), the code with `isSandbox=true` was added while fixing SECURITY-2008 issue in this commit: https://github.com/jenkinsci/active-choices-plugin/commit/62fb22b2d21d3bd2dfeaa051f283686c65dd6f7c

So definitely need care to merge it, so we don't re-introduce the security bug (which would end up removing the plug-in from update centre to all users)

Comment (kinow) [reply src/main/java/org/biouno/unochoice/model/ScriptlerScript.java:135]: Reading https://www.jenkins.io/doc/book/managing/script-approval/.

I think SECURITY-2008 because we were always evaluating the Groovy scripts in a non-sandbox environment and without approval.

At the moment we are always evaluating in a sandbox environment.

The change here would change it so that if a script has not been approved, then it would be sandboxed. But if already approved by an administrator, then the script would be evaluated without the Groovy sandbox.

I think it follows the model proposed in the Groovy Sandbox + Script Security, and doesn't add any security issues. So all good for iii) IMO.

**Comment by kinow** [comment]
Edited commit message, add changelog, added unit test. If CI passes, should be all good to be merged. Thanks @rafeeqpv !

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
Merged, thanks @rafeeqpv !

**Comment by daniel-beck** [comment]
@kinow Please hold off releasing for a bit. I'll contact you via Jira about this change.

**Comment by daniel-beck** [comment]
Consensus is that this should be an explicit choice as in other plugins, not implicit based on whether the script is approved.

**Comment by kinow** [comment]
Thanks @daniel-beck !

**Comment by kinow** [comment]
> @kinow Please hold off releasing for a bit. I'll contact you via Jira about this change.

I have another 4 or 5 issues that have been requested or have been in the backlog for a long long time. Unforunately not enough time to work on these issues, so this release might take a few more months (Oct/Nov with my current workload I guess).

**Pull request merged**
